src/hotspot/share/prims/jvmtiRedefineClasses.cpp

```diff
@@ -4403,7 +4403,9 @@ void VM_RedefineClasses::redefine_single_class(Thread* current, jclass the_jclas
     scratch_class->enclosing_method_method_index());
   scratch_class->set_enclosing_method_indices(old_class_idx, old_method_idx);
 
-  the_class->set_has_been_redefined();
+  if (!the_class->has_been_redefined()) {
+    the_class->set_has_been_redefined();
+  }
```

Comment:
Is this change really needed?

Reply:
This is needed by this change.  The has_been_redefined flag can be set at runtime and with these flags, there's an assert for all that they're only set once.  This one didn't have the assert but it can only be set in a safepoint, so I have this code to make it an exception. For now.  The plan is to make the _flags set once, and the _status field (to be added) set at runtime, and this can be moved to that.